Components/Terrain/include/OgreTerrain.i

```diff
@@ -0,0 +1,47 @@
+%module(package="Ogre") Terrain
+%{
+/* Includes the header in the wrapper code */
+#include "Ogre.h"
+#include "OgreTerrain.h"
+#include "OgreTerrainAutoUpdateLod.h"
+#include "OgreTerrainGroup.h"
+#include "OgreTerrainPagedWorldSection.h"
+#include "OgrePageManager.h"
+#include "OgrePagedWorld.h"
+#include "OgreTerrainPaging.h"
+// #include "OgreUnifiedHighLevelGpuProgram.h"
+%}
+
+%include std_string.i
+%include exception.i 
+%import "Ogre.i"
+
+#define _OgreTerrainExport
+
+%include "OgreTerrainLayerBlendMap.h"
```

Comment:
you must `%include "OgreTerrain.h"` first, so TerrainLayerBlendMap knows about it. This is what the error:
```
/home/runner/work/ogre/ogre/build/csharp/OgreTerrainCSHARP_wrap.cxx:450:3: error: ‘Terrain’ was not declared in this scope
   Terrain *arg1 = (Terrain *) 0 ;
   ^~~~~~~
/home/runner/work/ogre/ogre/build/csharp/OgreTerrainCSHARP_wrap.cxx:450:3: note: suggested alternative:
In file included from /home/runner/work/ogre/ogre/build/csharp/OgreTerrainCSHARP_wrap.cxx:383:0:
/home/runner/work/ogre/ogre/Components/Terrain/include/OgreTerrain.h:261:30: note:   ‘Ogre::Terrain’
```
means.

Reply:
Should I do a `%include` for every `#include`? I thought it wasn't necessary

Reply:
generally, it is not necessary. However %include is not recursive, so you might end up with missing declarations.